src/org/daisy/dotify/formatter/impl/volume/TocSequenceEventImpl.java

```diff
@@ -168,7 +169,13 @@ public BlockSequence getBlockSequence(FormatterContext context, DefaultContext v
                                 b.setMetaVolume(vol);
                             }
                             fsm.appendGroup(volumeStart);
-                            fsm.appendGroup(volumeToc);
+                            // If a chapter is so long that the there are two or
+                            // more resumed entries for that chapter in the
+                            // document TOC, the corresponding blocks must be
+                            // copied so that dynamic content (in particular
+                            // page numbers) is evaluated correctly. To avoid
+                            // code complexity, we simply copy all blocks here.
+                            fsm.appendGroup(volumeToc.stream().map(Block::copy).collect(Collectors.toList()));
```

Comment:
Can you try if you can make it work by only copying the `Block` objects that match a `toc-entry-on-resumed`? That is, the `Block` objects returned by `TableOfContentsImpl.filter` that are contained in `rangeForBlock`.

If it is done that way it becomes more obvious what the issue is and how we prevent it from happening: we want to avoid that the same `Block` object (with the same computed content) is rendered multiple times at different places. This can happen with `toc-entry-on-resumed` blocks and also `on-volume-start` etc. blocks (see the copy call in `getCompoundIterableB()`), but not with normal `toc-entry` blocks.

Reply:
@bertfrees Done. Please review again.

Reply:
I didn't know the block address also needed to be changed. But makes sense I guess. Complicates things a bit but it's OK.

Reply:
>I didn't know the block address also needed to be changed. But makes sense I guess. Complicates things a bit but it's OK.

I think I forgot that in the first case..